lib/puppet/provider/ec2_instance/v2.rb

```diff
@@ -46,10 +46,11 @@ def self.instance_to_hash(region, instance)
     end
     subnet_name = nil
     if instance.subnet_id
-      subnet_response = ec2_client(region).describe_subnets(subnet_ids: [instance.subnet_id])
-      subnet_name_tag = subnet_response.data.subnets.first.tags.detect { |tag| tag.key == 'Name' }
+      subnet_response = ec2_client(region).describe_subnets(filters: [
+        {name: 'subnet-id', values: [instance.subnet_id]}
+      ])
+      subnet_name = subnet_response.data.subnets.empty? ? nil : name_from_tag(subnet_response.data.subnets.first)
```

Comment:
Until Puppet 4, if we raise exceptions here they get eaten by Puppet. The execution continues with no prefetch data, meaning even stranger things happen at a distance. I've some ideas about how we can better handle that but this change definitely raises less issues from usage in my experience.